redfish-core/lib/pcie_slots.hpp

```diff
@@ -230,12 +242,166 @@ inline void handlePCIeSlotCollectionGet(
+    const std::string& chassisID, const std::string& chassisPath,
+    std::function<void(
+        const std::vector<std::pair<std::string, std::string>>&)>&& callback)
+{
+    BMCWEB_LOG_DEBUG("Get properties for PCIeSlots associated to chassis = {}",
+                     chassisID);
+
+    // get PCIeSlots that are in chassis
+    constexpr std::array<std::string_view, 1> pcieSlotIntf{
+        "xyz.openbmc_project.Inventory.Item.PCIeSlot"};
+
+    dbus::utility::getAssociatedSubTree(
+        chassisPath + "/inventory",
+        sdbusplus::message::object_path("/xyz/openbmc_project/inventory"), 0,
+        pcieSlotIntf,
+        [asyncResp, chassisID, chassisPath, callback{std::move(callback)}](
+            const boost::system::error_code& ec,
+            const dbus::utility::MapperGetSubTreeResponse& subtree) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("D-Bus response error on GetSubTree {}",
+                             ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        if (subtree.empty())
+        {
+            // No PCIeSlot found
+            messages::resourceNotFound(asyncResp->res, "Chassis", "PCIeSlot");
+            return;
+        }
+
+        std::vector<std::pair<std::string, std::string>> slotPathConnNames;
+        for (const auto& [pcieSlotPath, serviceName] : subtree)
+        {
+            if (pcieSlotPath.empty() || serviceName.size() != 1)
+            {
+                BMCWEB_LOG_ERROR("Error getting PCIeSlot D-Bus object!");
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            const std::string& connectionName = serviceName[0].first;
+            slotPathConnNames.emplace_back(
+                std::make_pair(pcieSlotPath, connectionName));
+        }
+
+        // sort by pcieSlotPath
+        std::sort(slotPathConnNames.begin(), slotPathConnNames.end(),
+                  [](const std::pair<std::string, std::string>& slot1,
+                     const std::pair<std::string, std::string>& slot2) {
+            return slot1.first < slot2.first;
+        });
+
+        callback(std::move(slotPathConnNames));
+    });
+}
+
+inline void doHandlePCIeSlotPatch(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp, size_t total,
+    const std::string& chassisID, const std::string& validChassisPath,
+    const std::map<unsigned int, bool>& locationIndicatorActiveMap)
+{
+    getValidPCIeSlotList(
+        asyncResp, chassisID, validChassisPath,
+        [asyncResp, chassisID, total,
+         locationIndicatorActiveMap{std::move(locationIndicatorActiveMap)}](
+            const std::vector<std::pair<std::string, std::string>>&
+                slotPathConnNames) {
+        if (slotPathConnNames.size() != total)
+        {
+            BMCWEB_LOG_WARNING(
+                "The actual number of pcieslots is different from the number of the input slots");
```

Comment:
Should 'pcieslots" here have some capital letters to match use elsewhere? (I.e. PCIeSlots)

Reply:
Changed.